cpp/src/arrow/ipc/memory.cc

```diff
@@ -145,5 +145,29 @@ Status MemoryMappedSource::Write(int64_t position, const uint8_t* data, int64_t
   return Status::OK();
 }
 
+MockMemorySource::MockMemorySource(int64_t size) : size_(size), pos_(0) {}
+
+Status MockMemorySource::Close() {
+  return Status::OK();
+}
+
+Status MockMemorySource::ReadAt(
+    int64_t position, int64_t nbytes, std::shared_ptr<Buffer>* out) {
+  return Status::OK();
+}
+
+Status MockMemorySource::Write(int64_t position, const uint8_t* data, int64_t nbytes) {
+  pos_ = std::max(pos_, position + nbytes);
```

Comment:
I think this is a variable naming and documentation problem. Can you change the variable name to `extent_bytes_written_` or something similar and add a comment to `Position` (or rename Position) to indicate that it returns the smallest number of contains containing the modified region of the MockMemorySource? Thanks